Gulpfile.ts

```diff
@@ -0,0 +1,1125 @@
+    "es2015.promise.d.ts",
+    "es2015.proxy.d.ts",
+    "es2015.reflect.d.ts",
+    "es2015.symbol.d.ts",
+    "es2015.symbol.wellknown.d.ts"
+];
+
+const es2015LibrarySourceMap = es2015LibrarySources.map(function(source) {
+   return  { target: "lib." + source, sources: ["header.d.ts", source] };
+});
+
+const es2016LibrarySource = [ "es2016.array.include.d.ts" ];
+
+const es2016LibrarySourceMap = es2016LibrarySource.map(function (source) {
+    return { target: "lib." + source, sources: ["header.d.ts", source] };
+});
+
+const es2017LibrarySource = [
+    "es2017.object.d.ts",
+    "es2017.sharedmemory.d.ts"
+];
+
+const es2017LibrarySourceMap = es2017LibrarySource.map(function (source) {
+    return { target: "lib." + source, sources: ["header.d.ts", source] };
+});
+
+const hostsLibrarySources = ["dom.generated.d.ts", "webworker.importscripts.d.ts", "scripthost.d.ts"];
+
+const librarySourceMap = [
+        // Host library
+        { target: "lib.dom.d.ts", sources: ["header.d.ts", "dom.generated.d.ts"] },
+        { target: "lib.dom.iterable.d.ts", sources: ["header.d.ts", "dom.iterable.d.ts"] },
+        { target: "lib.webworker.d.ts", sources: ["header.d.ts", "webworker.generated.d.ts"] },
+        { target: "lib.scripthost.d.ts", sources: ["header.d.ts", "scripthost.d.ts"] },
+
+        // JavaScript library
+        { target: "lib.es5.d.ts", sources: ["header.d.ts", "es5.d.ts"] },
+        { target: "lib.es2015.d.ts", sources: ["header.d.ts", "es2015.d.ts"] },
+        { target: "lib.es2016.d.ts", sources: ["header.d.ts", "es2016.d.ts"] },
+        { target: "lib.es2017.d.ts", sources: ["header.d.ts", "es2017.d.ts"] },
+
+        // JavaScript + all host library
+        { target: "lib.d.ts", sources: ["header.d.ts", "es5.d.ts"].concat(hostsLibrarySources) },
+        { target: "lib.es6.d.ts", sources: ["header.d.ts", "es5.d.ts"].concat(es2015LibrarySources, hostsLibrarySources, "dom.iterable.d.ts") }
+].concat(es2015LibrarySourceMap, es2016LibrarySourceMap, es2017LibrarySourceMap);
+
+const libraryTargets = librarySourceMap.map(function (f) {
+    return path.join(builtLocalDirectory, f.target);
+});
+
+for (const i in libraryTargets) {
+    const entry = librarySourceMap[i];
+    const target = libraryTargets[i];
+    const sources = [copyright].concat(entry.sources.map(function (s) {
+        return path.join(libraryDirectory, s);
+    }));
+    gulp.task(target, false, [], function() {
+        if (!needsUpdate(sources, target)) {
+            return gulp.src(target);
+        }
+        return gulp.src(sources).pipe(concat(target, {newLine: ""})).pipe(gulp.dest("."));
+    });
+}
+
+const configureNightlyJs = path.join(scriptsDirectory, "configureNightly.js");
+const configureNightlyTs = path.join(scriptsDirectory, "configureNightly.ts");
+const packageJson = "package.json";
+const programTs = path.join(compilerDirectory, "program.ts");
+
+function needsUpdate(source: string | string[], dest: string | string[]): boolean {

```

Comment:
The only place it's in use where streams actually get used is for concatenating the lib (everywhere else it's used before shelling out). However, I could probably change the lib creation tasks to use `gulp-newer` or `gulp-changed`.